bin/client/src/kona.rs

```diff
@@ -60,8 +60,8 @@ fn main() -> Result<()> {
         let mut executor = StatelessL2BlockExecutor::new(
             &boot.rollup_config,
             driver.take_l2_safe_head_header(),
+            l2_provider.clone(),
```

Comment:
@clabby My only question here is if we're confident enough in this "provider owns oracle" pattern to avoid the clone by changing this function to just take in a `provider` instead of both a `fetcher` and `hinter`?

It passes them to another function further downstream, so your call whether we want to make that change across the board or just use this clone and leave the other functions as is.